Dining/models.py

```diff
@@ -274,17 +231,48 @@ def clean(self):
                 })
 
             # Validate user is not already subscribed for the dining list
-            if not self.is_external() and self.dining_list.internal_dining_entries().filter(user=self.user).exists():
+            if self.get_internal() and self.dining_list.internal_dining_entries().filter(user=self.user).exists():
                 raise ValidationError(_('This user is already subscribed to the dining list.'))
 
             # (Optionally) validate if user is not already on another dining list
             #if DiningList.objects.filter(date=self.dining_list.date, dining_entries__user=self.user)
 
-    def __str__(self):
-        return str(self.user) + " " + str(self.dining_list.date)
+    def get_internal(self):
+        try:
+            return self.diningentryuser
+        except ObjectDoesNotExist:
+            return None
+
+    def get_external(self):
+        try:
+            return self.diningentryexternal
+        except ObjectDoesNotExist:
+            return None
+
+    def name(self):
+        if self.get_external():
+            return self.get_external().name
+        else:
+            return self.user
+
+class DiningWork(models.Model):
+    # Define the unique id name to prevent conflicts with DiningEntry
+    w_id = models.AutoField(primary_key=True)
+
+    # Add the stats
+    has_shopped = models.BooleanField(default=False)
+    has_cooked = models.BooleanField(default=False)
+    has_cleaned = models.BooleanField(default=False)
```

Comment:
Why not in `DiningEntryUser`? It's hard for me to follow what will be the end result in this case, I assume separate tables for `DiningWork` and `DiningEntryUser` which would result in more lookups.

Reply:
We can merge it, I have no preference, I figured that it it deviated slightly from the goal of a DiningEntry and could therefore be placed in a different table, should we ever need to make entries uncorrelated to the dining list.